lib/apiservers/engine/backends/container.go

```diff
@@ -415,6 +419,44 @@ func (c *Container) ContainerRm(name string, config *types.ContainerRmConfig) er
 	return nil
 }
 
+// checkActivePortBindings gets port bindings for the container and
+// unmaps ports if the cVM that previously bound them isn't powered on
+func (c *Container) checkActivePortBindings(vc *viccontainer.VicContainer) error {
+	for ctrPort, hostPorts := range vc.HostConfig.PortBindings {
+		for _, hostPort := range hostPorts {
+			hPort := hostPort.HostPort
+
+			cbpLock.Lock()
```

Comment:
You need to grab the lock for the duration of the outer for loop. This map is changed elsewhere when a port binding is created or removed, so the unmap code below could potentially clash with that.